administrator/components/com_workflow/src/Model/StagesModel.php

```diff
@@ -200,4 +202,74 @@ public function getWorkflow()
 
         return (object) $table->getProperties();
     }
+
+    /**
+     * Update positions for multiple workflow stages
+     *
+     * @param   array  $stages  Array of stage data, each with id, x, y values
+     *                          Example: [['id' => 1, 'x' => 100, 'y' => 200], ...]
+     *
+     * @return  boolean  True on success, false on failure
+     *
+     * @since   4.0.0
+     */
+    public function updatePositions($stagePositions)
```

Comment:
Probably worth to have a 2nd parameter "workflowId" so not random stages can be updated. Or this should be part of the controller validation. (also valid structure check)